packages/native-stack/src/types.tsx

```diff
@@ -342,6 +342,15 @@ export type NativeStackNavigationOptions = {
    * @platform ios
    */
   headerBackButtonMenuEnabled?: boolean;
+  /**
+   * How the back button behaves by default (when not customized). Available on iOS>=14.
+   * The following values are currently supported (they correspond to https://developer.apple.com/documentation/uikit/uinavigationitembackbuttondisplaymode?language=objc):
+   * - "default" – shows given back button title/previous controller title, system default or just icon based on available space
+   * - "generic" – shows given system default or just icon based on available space
+   * - "minimal" – shows just an icon
```

Comment:
We also need to make the descriptions more clear.

- The terminology should be updated (e.g. screen title instead of controller title)
- Not sure what "given" means in system default context, maybe a mistake?
- And add some example app what system default means in this case
- There being a "default" option and the "system default" also seem a bit confusing since they seem like different things